Sources/RelayerFramework/Services/EthereumService.swift

```diff
@@ -0,0 +1,161 @@
+            getBalance(address: address) { result in
+                switch result {
+                case let .success(balance):
+                    let messageResponse = Message(
+                        proto: UBID(repeating: 1, count: 1),
+                        recipient: message.origin,
+                        from: message.recipient,
+                        origin: message.recipient,
+                        message: balance
+                    )
+                    node.send(messageResponse)
+                case let .failure(error):
+                    print(error)
+                }
+            }
+        case "1":
+            let address = String(data: message.message[1 ..< message.message.count], encoding: .utf8)!
+            getTransactionCount(address: address) { result in
+                switch result {
+                case let .success(balance):
+                    let messageResponse = Message(
+                        proto: UBID(repeating: 1, count: 1),
+                        recipient: message.origin,
+                        from: message.recipient,
+                        origin: message.recipient,
+                        message: balance
+                    )
+                    node.send(messageResponse)
+                case let .failure(error):
+                    print(error)
+                }
+            }
+        case "2":
+            let signedTransaction = String(data: message.message[1 ..< message.message.count], encoding: .utf8)!
+            sendRawTransaction(signedTransaction: signedTransaction) { result in
+                switch result {
+                case let .success(balance):
+                    let messageResponse = Message(
+                        proto: UBID(repeating: 1, count: 1),
+                        recipient: message.origin,
+                        from: message.recipient,
+                        origin: message.recipient,
+                        message: balance
+                    )
+                    node.send(messageResponse)
+                case let .failure(error):
+                    print(error)
+                }
+            }
+            return
+        default:
+            print("default")
+            return
+        }
+    }
+
+    internal func httpRequest(request: URLRequest, completion: @escaping (Result<Data, Error>) -> Void) {
+        _ = URLSession.shared.dataTask(with: request) { data, _, error in
+            guard let data = data, error == nil else {
+                print(error?.localizedDescription ?? "No data")
+                completion(.failure(error!))
+                return
+            }
+            completion(.success(data))
+        }.resume()
+    }
+
+    internal func getBalance(address: String, completion: @escaping (Result<Data, Error>) -> Void) {
+        var request = URLRequest(url: url)
+        request.httpMethod = "POST"
+        let payload: [String: Any] = [
```

Comment:
payload construction could be done in a function where you pass the method and params and it returns the json serialized payload.